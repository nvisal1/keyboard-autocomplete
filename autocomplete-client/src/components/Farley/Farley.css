.farley__image-container {
    height: 200px;
}

@media only screen and (max-width: 786px) {
    .farley__image-container {
        height: 150px;
    }

    .farley__speech-bubble-container {
        position: absolute;
        bottom: 65px;
        left: 130px;

    }
}
