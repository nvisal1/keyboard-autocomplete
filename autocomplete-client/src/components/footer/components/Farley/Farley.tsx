import React from 'react';

interface FarleyProps {
    option: string,
}

const Farley: React.FC<FarleyProps> = props => {
    return (
        <div className='farley'>
       
        </div>
    );
}

export default Farley;