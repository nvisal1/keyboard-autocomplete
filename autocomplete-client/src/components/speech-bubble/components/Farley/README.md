## Farley

### Description

The beloved mascot! The Farley component provides an appropriately-sized image of Farley.

### Interface

This component does not accept props

### Parent Components

#### Speech Bubble

README at ../../README.md




