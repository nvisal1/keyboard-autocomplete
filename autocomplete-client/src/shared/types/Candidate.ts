export type Candidate = {
    word: string;
    confidence: number;
}