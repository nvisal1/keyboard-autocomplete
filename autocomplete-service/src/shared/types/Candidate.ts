export class Candidate {

    private _word: string;
    private _confidence: number;

    constructor(word: string, confidence: number) {
        this._word = word;
        this._confidence = confidence;
    }

    getWord(): string {
        return this._word;
    }

    getConfidence(): number {
        return this._confidence;
    }
}