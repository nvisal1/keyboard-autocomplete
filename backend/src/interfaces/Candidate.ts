export interface Candidate {
    getWord() : Promise<string>; //returns the autocomplete candidate
    getConfidence() : Promise<number> // returns the confidence* for the candidate
}