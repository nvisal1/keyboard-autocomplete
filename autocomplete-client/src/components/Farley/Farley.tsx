import React from 'react';
import FarleyImage from '../../shared/images/Farley.png';
import SpeechBubble from './components/speech-bubble/speech-bubble';
import './Farley.css';
import { Candidate } from '../../shared/types/Candidate';

interface FarleyProps {
    handleClick: Function;
    candidates: Candidate[];
}

const Farley: React.FC<FarleyProps> = props => {
    return (
        <div className='farley'>
            <img className='farley__image-container' src={FarleyImage} alt='Farley'/>
            <div className='farley__speech-bubble-container'>
                <SpeechBubble 
                    candidates={ props.candidates }
                    handleClick={ props.handleClick }
                ></SpeechBubble>
            </div>
        </div>
    );
}

export default Farley;