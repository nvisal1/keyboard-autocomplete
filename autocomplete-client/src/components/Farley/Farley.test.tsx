import React from 'react';
import renderer from 'react-test-renderer';

import Farley, { FarleyProps } from './Farley';
import { Candidate } from '../../shared/types/Candidate';

describe('When the Farley component is created', () => {
    describe('and the errorMessage prop is set to null', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps();
            const currentSnapshot = renderer.create(<Farley { ...requiredProps }></Farley>).toJSON();

            expect(currentSnapshot).toMatchSnapshot();
        });
    });
    describe('and the errorMessage prop is set to a truthy string', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps();
            const propsWithErrorMessage = { ...requiredProps, errorMessage: 'test_error_messagge' };
            const currentSnapshot = renderer.create(<Farley { ...propsWithErrorMessage }></Farley>).toJSON();

            expect(currentSnapshot).toMatchSnapshot();
        });
    });
});

function getRequiredProps(): FarleyProps {
   const handleClick = (_: string) => {}; 

   const candidates: Candidate[] = [{ word: 'test_word', confidence: 1}]; 

   return { handleClick, candidates };
}