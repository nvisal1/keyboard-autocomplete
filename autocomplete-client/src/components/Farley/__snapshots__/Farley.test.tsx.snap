// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the Farley component is created and the errorMessage prop is set to a truthy string should render correctly 1`] = `
<div
  className="farley"
>
  <img
    alt="Farley"
    className="farley__image-container"
    src="Farley.png"
  />
  <div
    className="farley__speech-bubble-container"
  >
    <div
      className="speech-bubble"
    >
      <div
        className="speech-bubble__text-container"
      >
        <p
          className="speech-bubble__text-container__error-message"
        >
          test_error_messagge
        </p>
        <div
          className="speech-bubble__pointer-container"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`When the Farley component is created and the errorMessage prop is set to null should render correctly 1`] = `
<div
  className="farley"
>
  <img
    alt="Farley"
    className="farley__image-container"
    src="Farley.png"
  />
  <div
    className="farley__speech-bubble-container"
  >
    <div
      className="speech-bubble"
    >
      <div
        className="speech-bubble__text-container"
      >
        <p
          className="speech-bubble__text-container__word"
          onClick={[Function]}
        >
          test_word(1)
        </p>
        <div
          className="speech-bubble__pointer-container"
        />
      </div>
    </div>
  </div>
</div>
`;
